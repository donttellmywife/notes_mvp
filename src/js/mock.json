{
  "notes": [
    {
      "id": 6,
      "bookId": 4,
      "name": "benchmark leading-edge Colorado",
      "text": "Velit corporis autem aspernatur. Fugit suscipit quae dolor occaecati atque eveniet nihil molestias id. Quas sit veniam voluptas optio. Itaque harum aperiam perferendis qui ex. Totam asperiores tenetur. Aut error incidunt molestiae eaque hic impedit.\n \rQuia mollitia in maxime. Esse earum sequi consequatur odio tempore. Aperiam illum fugiat porro ea dolorem vel assumenda ullam pariatur. Ut error voluptas. Recusandae eligendi unde tempora deleniti nemo delectus.\n \rDolorum quibusdam possimus velit voluptatibus atque. Expedita rem error ad reiciendis quia iusto. Modi deleniti numquam quo rem eum placeat reiciendis illo ea. Porro consectetur voluptates quae. Minus odit at et optio quibusdam adipisci. Dolorem vel doloribus."
    },
    {
      "id": 7,
      "bookId": 4,
      "name": "HDD array",
      "text": "Dolor rerum ipsam quis velit. Non laborum quia qui. Illo et autem quidem.\n \rEos qui distinctio distinctio voluptate voluptas nemo mollitia esse. Optio at voluptates earum. Labore consequatur et quia tempora voluptatibus sint. Consequuntur voluptas vitae aut quidem eum fugit est aperiam. Iusto architecto rerum reiciendis ea in aut cupiditate. Sunt praesentium nostrum aut aspernatur rerum.\n \rDelectus ea quasi ad officia. Ea qui provident. Nihil eveniet iusto odit et."
    }
  ]
}
