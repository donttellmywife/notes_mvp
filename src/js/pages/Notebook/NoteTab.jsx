import React from 'react'
import { string, number, func } from 'prop-types'


function NoteTab({name, className, handleClick = noop}) {
  const cls = `tab ${className}`

  return (
    <div onClick={handleClick} className={cls}>
      <h1>{`${name}`}</h1>
    </div>
  )
}

NoteTab.defaultProps = {
  handleClick: noop
}
NoteTab.propTypes = {
  id: number.isRequired,
  name: string.isRequired,
  handleClick: func
}

function noop() {
  return undefined;
}

export default NoteTab
