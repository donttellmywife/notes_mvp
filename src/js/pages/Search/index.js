export { default as Search } from './Search'
export { default as searchTerm } from './redux/r'
export { default as setSearchTerm } from './redux/ac'
