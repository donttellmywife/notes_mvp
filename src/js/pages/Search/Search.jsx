import React from 'react'
import { connect } from 'react-redux'
import { string, array } from 'prop-types'

import NoteTab from '../Notebook/NoteTab'
import { Header } from '../../components'

const Search = ({ searchTerm, items }) => {
  const notes = items
    .filter((note) => `${note.name} ${note.text}`.toUpperCase()
      .indexOf(searchTerm.toUpperCase()) >= 0
    )
    .map((note) => <NoteTab key={note.id} {...note}/>)

  return (
    <div className="search">
      <Header showSearch />
      {notes}
    </div>
  )
}

Search.propTypes = {
  searchTerm: string,
  items: array,
}
Search.defaultProps = {
  searchTerm: '',
  items: [],
}

const mapStateToProps = (state) => ({
  searchTerm: state.searchTerm
})

export const Unwrapped = Search
export default connect(mapStateToProps)(Search)
