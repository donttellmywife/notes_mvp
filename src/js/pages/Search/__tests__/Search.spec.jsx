import React from 'react'
import Enzyme, { shallow }  from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })

import Search, { Unwrapped as UnwrappedSearch } from '../Search'
import NoteTab from '../../Notebook/NoteTab'

import mock from './mock.json'


describe('Search page: ', () => {

  it('Search renders correctrly', () => {
    const component = shallow(<UnwrappedSearch items={mock.notes} searchTerm="" />)
    expect(component).toMatchSnapshot()
  })


  it('Search should render correct amount of notes', () => {
    const component = shallow(<UnwrappedSearch items={mock.notes} searchTerm="" />)
    expect(component.find(NoteTab).length).toEqual(mock.notes.length)
  })


  it('Search should render correct amount of notes based on search input', () => {
    const search = 'leading'
    const component = shallow(<UnwrappedSearch items={mock.notes} searchTerm={search} />)
    const noteCount = mock.notes
      .filter((note) => `${note.name} ${note.text}`.toUpperCase()
        .indexOf(search.toUpperCase()) >= 0
      ).length

    expect(component.find(NoteTab).length).toEqual(noteCount)
  })
})
