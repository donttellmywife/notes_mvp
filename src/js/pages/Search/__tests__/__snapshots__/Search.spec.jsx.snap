// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search page:  Search renders correctrly 1`] = `
<div
  className="search"
>
  <Connect(Header)
    showSearch={true}
  />
  <NoteTab
    bookId={4}
    handleClick={[Function]}
    id={6}
    key="6"
    name="benchmark leading-edge Colorado"
    text="Velit corporis autem aspernatur. Fugit suscipit quae dolor occaecati atque eveniet nihil molestias id. Quas sit veniam voluptas optio. Itaque harum aperiam perferendis qui ex. Totam asperiores tenetur. Aut error incidunt molestiae eaque hic impedit.
 
Quia mollitia in maxime. Esse earum sequi consequatur odio tempore. Aperiam illum fugiat porro ea dolorem vel assumenda ullam pariatur. Ut error voluptas. Recusandae eligendi unde tempora deleniti nemo delectus.
 
Dolorum quibusdam possimus velit voluptatibus atque. Expedita rem error ad reiciendis quia iusto. Modi deleniti numquam quo rem eum placeat reiciendis illo ea. Porro consectetur voluptates quae. Minus odit at et optio quibusdam adipisci. Dolorem vel doloribus."
  />
  <NoteTab
    bookId={4}
    handleClick={[Function]}
    id={7}
    key="7"
    name="HDD array"
    text="Dolor rerum ipsam quis velit. Non laborum quia qui. Illo et autem quidem.
 
Eos qui distinctio distinctio voluptate voluptas nemo mollitia esse. Optio at voluptates earum. Labore consequatur et quia tempora voluptatibus sint. Consequuntur voluptas vitae aut quidem eum fugit est aperiam. Iusto architecto rerum reiciendis ea in aut cupiditate. Sunt praesentium nostrum aut aspernatur rerum.
 
Delectus ea quasi ad officia. Ea qui provident. Nihil eveniet iusto odit et."
  />
</div>
`;
